### PR TITLE
Add toggle to mark a dataObject as a collection

### LINKED ISSUE #381: I can model data object markers
AT:
- collection
- input/output data objects

### PR DESCRIPTION
**Pull request by azeghers**
__Which issue does this PR address?__

Closes #381

__Breakdown__

* [X] Feature is implemented
* [x] Undo and redo are possible
* [x] Tests are added

__Acceptance Criteria__

* [ ] Corresponds to the concept 
* [ ] Corresponds to the design 

__Definition of Done__

* [ ] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [ ] corresponds to [the code standards](https://github.com/bpmn-io/bpmn-js/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [ ] passes Continuous Integration checks
* [ ] available as feature branch on GitHub
  * [ ] contains the cleaned up commit history
  * [ ] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [ ] a single commit closes the issue via Closes #issuenr

### CODE REVIEW

File: lib/features/modeling/Modeling.js [hunk 1]
```diff
@@ -44,15 +45,16 @@ Modeling.$inject = [
 Modeling.prototype.getHandlers = function() {
   var handlers = BaseModeling.prototype.getHandlers.call(this);
 
-  handlers['element.updateProperties'] = UpdatePropertiesHandler;
```
Comment (nikku) [inline lib/features/modeling/Modeling.js:47]: What happened here?

File: lib/features/modeling/Modeling.js [hunk 2]
```diff
@@ -2,14 +2,15 @@ import inherits from 'inherits';
 
 import BaseModeling from 'diagram-js/lib/features/modeling/Modeling';
 
-import UpdatePropertiesHandler from './cmd/UpdatePropertiesHandler';
```
Comment (nikku) [inline lib/features/modeling/Modeling.js:5]: What happened here?

File: lib/features/modeling/cmd/UpdateDataObjectHandler.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+var reduce = require('min-dash').reduce,
+    keys = require('min-dash').keys,
+    forEach = require('min-dash').forEach,
+    is = require('../../../util/ModelUtil').is,
+    getBusinessObject = require('../../../util/ModelUtil').getBusinessObject;
+
+
+function UpdateDataObjectHandler(elementRegistry) {
```
Comment (nikku) [inline lib/features/modeling/cmd/UpdateDataObjectHandler.js:10]: :+1: for the complete solution, updating all referenced data object references.

File: test/spec/features/context-pad/ContextPadProviderSpec.js [hunk 4]
```diff
@@ -489,7 +489,7 @@ describe('features - context-pad', function() {
 
           // given
           var rootShape = canvas.getRootElement(),
-              dataObject = elementFactory.createShape({ type: 'bpmn:DataObjectReference' }),
+              dataObject = elementFactory.createShape({ type: 'bpmn:DataStoreReference' }),
```
Comment (nikku) [inline test/spec/features/context-pad/ContextPadProviderSpec.js:492]: This does not seem right: 

```
dataObject = elementFactory.createShape({ type: 'bpmn:DataStoreReference' })
```

Comment (azeghers) [reply test/spec/features/context-pad/ContextPadProviderSpec.js:492]: True, it was originally a dataObject but the test was failing now that dataObjectReferences can have a replace menu.
I forgot to rename the variable as well.

File: lib/features/modeling/Modeling.js (hunk 1 above)
Comment (azeghers) [reply lib/features/modeling/Modeling.js:47]: Alphabetical sorting, it's a habit of mine but I didn't intend to commit it in the end, my bad.

Comment (nikku) [reply lib/features/modeling/Modeling.js:47]: Fine to do these kind of things, just put them in an additional "chore" or "style" commit.

File: test/spec/features/popup-menu/ReplaceMenuProviderSpec.js [hunk 5]
```diff
@@ -49,6 +50,144 @@ describe('features/popup-menu - replace menu provider', function() {
     });
   };
 
+  describe('data object - collection marker', function() {
```
Comment (nikku) [inline test/spec/features/popup-menu/ReplaceMenuProviderSpec.js:53]: As mentioned earlier today, we should split up the UI specific tests from the ones that validate the API (`Modeling#...`) usage and `commandStack.changed` semantics.

**Comment by azeghers** [comment]
```
16 11 2020 13:03:00.533:WARN [karma]: No captured browser, open http://localhost:9876/
16 11 2020 13:03:00.546:INFO [karma-server]: Karma v4.4.1 server started at http://0.0.0.0:9876/
16 11 2020 13:03:00.546:INFO [launcher]: Launching browsers PhantomJS with concurrency unlimited
16 11 2020 13:03:00.550:INFO [launcher]: Starting browser PhantomJS
16 11 2020 13:03:00.688:INFO [PhantomJS 2.1.1 (Linux 0.0.0)]: Connected on socket CRchEO9S5hSfER6xAAAA with id 73262039
PhantomJS 2.1.1 (Linux 0.0.0) ERROR
  SyntaxError: Unexpected token '>'
  at test/testBundle.js:4251:0
```

I also have this very weird parsing error that only disappears if I don't import { spy } from 'sinon';

**Comment by nikku** [comment]
> I also have this very weird parsing error that only disappears if I don't import { spy } from 'sinon';

`{ spy }` is ES6. The target runtime (PhantomJS) supports ES5 only.

Please use `import('foo').spy` instead and you're o.k.

**Comment by azeghers** [comment]
But the rest of the file imports in an ES6 way
image

And importing via `import()` or `require()` doesn't seem to change anything.
image

The tests run fine in the browser, but PhantomJS just throws in the terminal

**Comment by nikku** [comment]
Never mind my previous comment. 

Sinon is baked in via `karma-sinon-chai`. It is available as a global in the tests, simlar to `describe` and `it`.

**Comment by barmac** [comment]
The code looks good, but the marker is a bit off. 

image

Compare it with spec:

image

**Review by barmac — APPROVED** [review]

**Pull request merged**